### PR TITLE
[ETP-51] 138725 work even if post only has RSVPs

### PR DESCRIPTION
**Pull request by cliffordp**
https://central.tri.be/issues/138725

4min demo video: https://share.getcloudapp.com/6quLQloN

*See question comments, below...*

[ETP-51]

[ETP-51]: https://moderntribe.atlassian.net/browse/ETP-51

### CODE REVIEW

**Comment by cliffordp** [comment]
@sc0ttkclark 
1. If an event has both RSVPs and Tickets, it currently anchor links to Tickets, which displays below RSVPs. Is this the way we want it? (see demo video)
1. Do we want to remove the exclamation marks from "Buy Now!" and "RSVP Now!" button text?

@Camwyn 
1. The use of `<form>` and `<button>` was a way to get a link that looks like a button. I tried `a.tribe-button` without a form or button but that class alone doesn't style it the same as `button.tribe-button` (due to Twenty Twenty's theme CSS styling of `button` elements from `<style id="twentytwenty-style-inline-css">...`)
    1. 1min demo with a site using "Tribe Events" styles (not Full nor Skeleton): https://share.getcloudapp.com/BluNYwQ1
    1. Is `a.tribe-button` lacking styling or do we need to keep it as a button-form?
    1. If we _have_ to keep button-form, should we add `value="1"` or fully drop the button's `name="tickets_process"`?

**Comment by cliffordp** [comment]
@sc0ttkclark checking in...

**Comment by Camwyn** [comment]
>     1. The use of `<form>` and `<button>` was a way to get a link that looks like a button. I tried `a.tribe-button` without a form or button but that class alone doesn't style it the same as `button.tribe-button` (due to Twenty Twenty's theme CSS styling of `button` elements from `<style id="twentytwenty-style-inline-css">...`)
>        
>        1. 1min demo with a site using "Tribe Events" styles (not Full nor Skeleton): https://share.getcloudapp.com/BluNYwQ1 -- and a 1.5min video wondering why `tribe-ui` CSS file is not helping render `.tribe-button`: https://share.getcloudapp.com/yAuL6NNr
>        2. Is `a.tribe-button` lacking styling or do we need to keep it as a button-form?
>        3. If we _have_ to keep button-form, should we add `value="1"` or fully drop the button's `name="tickets_process"`?
> 
> 
> [cbbe106](https://github.com/moderntribe/event-tickets/commit/cbbe1068a5696863bea247d1bbb2891aba931e20) is the commit to demonstrate the link instead of button-form. I reverted it to keep this commit for your reference until I hear back a decision.

The new styles use different classes:
`tribe-common-c-btn` is the button class you want.
For it to work it needs to be _inside_ a container with the `tribe-common` class - which should probably be the loop, but check with Blue on that?

**Comment by cliffordp** [comment]
@Camwyn thanks for confirming it's okay to not be in a `<form>` and thanks for the tips :)

File: src/template-tags/tickets.php [hunk 1]
```diff
@@ -181,31 +181,31 @@ function tribe_events_count_available_tickets( $event = null ) {
...
 	 * @since  4.5
+	 * @since  TBD Now displays for posts having only RSVPs.
 	 *
 	 * @param bool $echo Whether or not we should print
 	 *
 	 * @return string
 	 */
 	function tribe_tickets_buy_button( $echo = true ) {
-		$event_id = get_the_ID();
+		$event = get_post();
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]: Do we need the full object in this function? If not, we can use `get_the_ID()` and bail if `false === $event_id`

Comment (cliffordp) [reply src/template-tags/tickets.php]: OK.

**Comment by cliffordp** [comment]
I changed base branch to `B20.01` to use Matt's new `tribe_tickets_is_current_time_in_date_window()` from https://github.com/moderntribe/event-tickets/pull/1528

**Comment by cliffordp** [comment]
@Camwyn I tried the classes but no 🎲 
Any ideas how to fix? https://share.getcloudapp.com/eDu95yj5 (2min)

@sc0ttkclark 
1. If an event has both RSVPs and Tickets, it currently anchor links to Tickets, which displays below RSVPs. Is this the way we want it? (see demo video)
1. Do we want to remove the exclamation marks from "Buy Now!" and "RSVP Now!" button text?

File: src/template-tags/tickets.php [hunk 2]
```diff
@@ -192,20 +193,24 @@ function tribe_events_count_available_tickets( $event = null ) {
...
 		$event_id = get_the_ID();
 
-		// check if there are any tickets on sale
-		if ( ! tribe_events_has_tickets_on_sale( $event_id ) ) {
-			return null;
+		if ( empty( $event_id ) ) {
+			return '';
+		}
+
+		// check if there are any tickets available
```
Comment (tr1b0t) [inline src/template-tags/tickets.php]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 05623631a75c14eacc1a1b5f4c8d5569</sub>

File: src/template-tags/tickets.php [hunk 3]
```diff
@@ -192,20 +193,24 @@ function tribe_events_count_available_tickets( $event = null ) {
...
 		$types = Tribe__Tickets__Tickets::get_ticket_counts( $event_id );
 
 		// if no rsvp or tickets return
 		if ( ! $types ) {
-			return null;
+			return '';
 		}
 
-		$html = [];
+		$html  = [];
```
Comment (tr1b0t) [inline src/template-tags/tickets.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 9deff1cc9527e8b82b96c96f8c5ad379</sub>

File: src/template-tags/tickets.php [hunk 4]
```diff
@@ -1428,4 +1421,4 @@ function tribe_tickets_is_enabled_post_context( $post = null ) {
 		 */
 		return apply_filters( 'tribe_tickets_is_enabled_post_context', false, $post_types, $context );
 	}
-}
+}
```
Comment (tr1b0t) [inline src/template-tags/tickets.php]: :no_entry_sign: File must end with a newline character

<sub>hash: 54e1664c1d243f82e819937d62bc2769</sub>

Comment (cliffordp) [reply src/template-tags/tickets.php]: @borkweb I thought it best practice _not_ to have empty/new line at the end of PHP files?

File: src/template-tags/tickets.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/template-tags/tickets.php]:
```suggestion
		// Check if there are any tickets available.
```

File: src/template-tags/tickets.php [hunk 5]
```diff
@@ -239,11 +244,12 @@ function tribe_tickets_buy_button( $echo = true ) {
...
 					 * Overwrites the threshold to display "# tickets left".
 					 *
-					 * @param int   $threshold Stock threshold to trigger display of "# tickets left"
+					 * @since 4.10.1
+					 *
 					 * @param array $data      Ticket data.
 					 * @param int   $event_id  Event ID.
 					 *
-					 * @since 4.10.1
+					 * @param int   $threshold Stock threshold to trigger display of "# tickets left"
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
					 * @param int   $threshold Stock threshold to trigger display of "# tickets left" text.
```

File: src/template-tags/tickets.php [hunk 6]
```diff
@@ -277,24 +283,11 @@ function tribe_tickets_buy_button( $echo = true ) {
...
-					// a query string must be 2 parts only a name and a value
-					if ( is_array( $parts ) && 2 === count( $parts ) ) {
-						list( $name, $value ) = $parts;
-						$button .= '<input type="hidden" name="' . esc_attr( $name ) . '" value="' . esc_attr( $value ) . '">';
-					}
-				}
-
-				$button	.= '<button type="submit" name="tickets_process" class="tribe-button">' . esc_html( $button_label ) . '</button></form>';
+				$button = sprintf(
+					'<div class="tribe-common"><a class="tribe-common-c-btn" href="%s"><button class="tribe-common-c-btn">%s</button></a></div>',
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
					'<div class="tribe-common"><a class="tribe-common-c-btn" href="%1$s"><button class="tribe-common-c-btn">%2$s</button></a></div>',
```

File: src/template-tags/tickets.php (hunk 4 above)
Comment (sc0ttkclark) [reply src/template-tags/tickets.php]:
```suggestion
}

```

Not sure there but we can just fix that here

**Comment by sc0ttkclark** [comment]
> If an event has both RSVPs and Tickets, it currently anchor links to Tickets, which displays below RSVPs. Is this the way we want it? (see demo video)

Not much we can do there, Tickets take priority in case here.

> Do we want to remove the exclamation marks from "Buy Now!" and "RSVP Now!" button text?

That's more of a strategy question, best to ask that in the ticket itself and @ Jeff there.

**Comment by sc0ttkclark** [comment]
This PR ended up merging into the wrong branch, Cliff will be remaking the branch based on `release/G20.01`

**Pull request merged**